composer.json

```diff
@@ -30,7 +30,7 @@
     },
     "require": {
         "php": "^7.3 || ~8.0.0",
-        "doctrine/doctrine-module": "^2.1.8 || ^3.0.1 || ^4.1.0",
+        "doctrine/doctrine-module": "^2.1.8 || ^3.0.1 || ^4.1.0 || ^5.1",
         "laminas-api-tools/api-tools-api-problem": "^1.2.2",
```

Comment:
I suggest we start dropping older versions: two majors is sufficient

Reply:
I tried, but once more we need to add newer versions then in other libraries:

`Problem 1
    - doctrine/doctrine-orm-module[2.1.3, ..., 2.1.7] require doctrine/doctrine-module ^2.1.3 -> found doctrine/doctrine-module[2.1.3, ..., 2.1.10] but it conflicts with your root composer.json require (^4.1.0 || ^5.1).
    - Root composer.json requires doctrine/doctrine-orm-module ^2.1.3 -> satisfiable by doctrine/doctrine-orm-module[2.1.3, ..., 2.1.7].`

Reply:
When I am starting to drop it I am facing this problems:
```
 Problem 1
    - Root composer.json requires laminas-api-tools/api-tools-doctrine ^2.1 -> satisfiable by laminas-api-tools/api-tools-doctrine[2.1.0, ..., 2.4.0].
    - Conclusion: don't install doctrine/doctrine-module 4.3.0 (conflict analysis result)
    - Conclusion: don't install doctrine/doctrine-module 4.3.1 (conflict analysis result)
    - Conclusion: don't install doctrine/doctrine-module 4.4.1 (conflict analysis result)
    - doctrine/doctrine-orm-module 5.1.0 requires doctrine/doctrine-module ^5.1.0 -> satisfiable by doctrine/doctrine-module[5.1.0].
    - Conclusion: don't install doctrine/doctrine-module 5.1.0 (conflict analysis result)
    - doctrine/doctrine-orm-module[4.1.0, ..., 4.2.1] require doctrine/doctrine-module ^4.2.2 -> satisfiable by doctrine/doctrine-module[4.2.2, ..., 4.4.1].
    - Conclusion: don't install doctrine/doctrine-module 4.2.2 (conflict analysis result)
    - Root composer.json requires doctrine/doctrine-orm-module ^4.1.0 || ^5.1 -> satisfiable by doctrine/doctrine-orm-module[4.1.0, 4.2.0, 4.2.1, 5.1.0].
```

Because there are same dev dependencies there:
https://github.com/laminas-api-tools/api-tools-doctrine/blob/2.5.x/composer.json#L42-L44

I am afraid if I fix in there, some other modules will have same problems.

Reply:
@Ocramius did the same here: https://github.com/laminas-api-tools/api-tools-doctrine/issues/28.